userspace/libsinsp/container_engine/docker_linux.cpp

```diff
@@ -195,7 +155,7 @@ docker::docker_response libsinsp::container_engine::docker::get_docker(sinsp_con
 	{
 		case 0: /* connection failed, apparently */
 			g_logger.format(sinsp_logger::SEV_NOTICE, "Docker connection failed, disabling Docker container engine");
-			m_enabled = false;
+			docker::set_enabled(false);
```

Comment:
I realize that this is just massaging what was already there, but you might consider moving this to `parse_docker`, where we handle the `RESP_ERROR` case.  It feels a little unnatural to have docker-level implementation details down in the bowels of HTTP fetching code.

In my original version of this change, I decoupled URL fetching logic from the container engine:

https://github.com/draios/sysdig/blob/f850d3b426d13eddaacb7d08023bb5127d745e0f/userspace/libsinsp/curl_url_fetcher.cpp

I'd still like to do that eventually since it'll enable us to more easily UT this code by injecting a mock realization of the fetcher that returns pre-canned responses.  If we can keep the URL fetching logic separate from the docker details, it'll help with that.

Reply:
Sure, I'll pass a failure all the way back to the callback and set m_enabled there.